portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,36 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  private UserService userService = UserServiceFactory.getUserService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    boolean status = userService.isUserLoggedIn();
+    response.getWriter().println(Boolean.toString(status));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    if (userService.isUserLoggedIn()) {
+      String urlToRedirectToAfterUserLogsOut = "/";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      response.getWriter().println(logoutUrl);
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+      response.getWriter().println(loginUrl);
+    }
+  }
+}
```

Comment:
POST should be used to create new resources (like adding a new entity to your database). GET should be used to get data without modifying data in the backend.
https://www.restapitutorial.com/lessons/httpmethods.html

Maybe you could get your GET to both things you want. You could make it return a JSON object with a boolean for whether they are logged in or not and also an action url (that would be either login or logout based on the other boolean).

Reply:
I was being lazy (didn't want to create a new class) and somehow managed to convince myself that "login" is to write the user's status (??). Now fixed lol